Post Video Renderer

aliseyalvi opened the pull request:
### What does this PR?
Added iframe support in Renderhtml by adding @native-html/iframe-plugin .
Added webview in anchor renderer to render markdown-video-link in webview instead of in a separate modal.


### Issue number
Trello: explore iframe native render plugin

### Screenshots

![image](https://user-images.githubusercontent.com/48380998/150473802-e8b3cc05-0c61-4ab0-a1e1-784eb708093d.png)

noumantahir commented:
@feruzm 
Current and new video playback system comparison....

Current system
https://user-images.githubusercontent.com/6298342/150485977-e75a0ce4-efbe-4964-ab17-e50027c22dd8.mov

New System in progress
https://user-images.githubusercontent.com/6298342/150485924-3415490a-7b63-4897-98c5-5ffb716c6dc8.mov

noumantahir commented:
@aliseyalvi please test and try to perfect the play back of this post, it's a perfect example of a post content containing iframes....

https://ecency.com/hive-172868/@vaketo/-----xgo102be5k8xjugq0drdwpsg4rjvu0gk

feruzm commented:
> @feruzm Current and new video playback system comparison....
> 
> Current system https://user-images.githubusercontent.com/6298342/150485977-e75a0ce4-efbe-4964-ab17-e50027c22dd8.mov
> 
> New System in progress https://user-images.githubusercontent.com/6298342/150485924-3415490a-7b63-4897-98c5-5ffb716c6dc8.mov

I like new system 👏

aliseyalvi commented:
@noumantahir 

- Added iframe support. Replaced iframe with custom iframe renderer using webview.  https://github.com/ecency/ecency-mobile/pull/2154/commits/68a94dd2c13541ab5c11b77430c98f66fc35f9d5
- Wrapped source with video tag if  source is video file which fixes the autoplay issue.
- changed webview source for supporting all type of videos like Youtube, vimeo etc. https://github.com/ecency/ecency-mobile/pull/2154/commits/8b4e2ea013c661cb4f03d881e255eb3fdadfd670
- Some Youtube videos are showing not available error. Might be due to privacy issue or could be other bug. 
![image](https://user-images.githubusercontent.com/48380998/150535931-a0c9240e-0e9a-4055-95e7-b525075a01a5.png)

noumantahir commented:
@aliseyalvi after assessing your solution, it seems we should be able to easily repurpose `videoPlayerSheet.tsx` to achieve our goal instead of rewriting webviews.

We can rename VideoPlayerSheet to simple VideoPlayer and call it with videoUrl for regular videos and with youtube id in case of youtube and that should also fix the youtube issue you are facing.

Another suggestion from my side would be to skip comments from using new video playback system as they can overload the system.

noumantahir commented on src/components/postHtmlRenderer/postHtmlRenderer.tsx:
```diff
@@ -137,7 +137,32 @@ export const PostHtmlRenderer = memo(({
+            allowsInlineMediaPlayback={true}
+          />
+        );
+      }
```
for youtube... the class is `markdown-video-link-youtube` and attributes are as follows

data-youtube: contains video id
data-start-time: contains start time of video (optional)

pass this data to youtube iframe implemented inside videoPlayerSheet.tsx and it should play without issues.

noumantahir reviewed (commented):
It's working alright, video source with extension is an exception we can accept for now.... Do checkout my comment regarding how we can repurpose videoPlayerSheet to avoid code duplications.

aliseyalvi commented:
@noumantahir created videoPlayer component which handles youtube and other videos as well. Iframe renderer has kept as it is because it handles source differently. review the changes https://github.com/ecency/ecency-mobile/pull/2154/commits/9c3f12e06e4b52dac0c3328382f12dbfc6f828bc

noumantahir commented:
**Updates**
Unified Video Player Component
Using Sheet Style player for comments and inline videos
Disabled autoplay youtube for post videos
Simplified anchorRenderer

@aliseyalvi please go through changes and test if I missed anything that was intentional
@feruzm you may review as well

noumantahir reviewed (approved):
All looks good to me now... require more testing and verifying.

noumantahir commented:
@feruzm this PR is awaiting your approval before merging

feruzm reviewed (approved):
lgtm
